### PR TITLE
Recaptcha integration

### LINKED ISSUE #8: Recaptcha integration for login and generation

### PR DESCRIPTION
**Pull request by abarghoud**
Close #8 

- New libraries to handle recaptcha token generation and validation
- API endpoint to verify a recaptcha token
- Protection of authentication actions with a recaptcha verification
- Protection of generation endpoint with a recaptcha verification
- Deactivation of recaptcha security locally

### CODE REVIEW

File: apps/api/.env.example [hunk 1]
```diff
@@ -0,0 +1,5 @@
+RECAPTCHA_SITE_KEY=
+FIREBASE_PROJECT_ID=
+GOOGLE_APPLICATION_CREDENTIALS=
+FRONTEND_URL=
+ENVIRONMENT=local
```
Comment (Romain-Portanguen) [inline apps/api/.env.example:5]: What is the point of specifying ‘local’ in the example?

File: apps/api/src/app/app.module.ts [hunk 2]
```diff
@@ -14,15 +14,23 @@ import { PPSApi } from './pps/third-party/pps-api';
 import { IPPSApiResponseAuthenticationMetaDataExtractorSymbol } from './pps/domain/authentication-metadata-extractor/pps-api-response-authentication-metadata-extractor.interface';
 import { PPSApiResponseAuthenticationMetaDataExtractor } from './pps/domain/authentication-metadata-extractor/pps-api-response-authentication-metadata-extractor';
 import { KeepAliveController } from './keep-alive/keep-alive.controller';
+import { IRecaptchaCheckerSymbol } from '@pps-easy/recaptcha/contracts';
+import { GoogleRecaptchaChecker, LocalRecaptchaChecker } from '@pps-easy/recaptcha/google';
+import * as process from 'node:process';
+import { RecaptchaController } from './recpatcha/recaptcha.controller';
+import { RecaptchaGuard } from './guards/recaptcha.guard';
 
+const isLocalEnvironment = process.env.ENVIRONMENT;
```
Comment (Romain-Portanguen) [inline apps/api/src/app/app.module.ts:23]: Hmm ok, I guess the value specified in the example is for this variable, right?

**Review by Romain-Portanguen — APPROVED** [review]
As usual... Impressive and more than clean. awsome 😍 
Just run the Lint scripts and we'll be good to go !

File: apps/api/.env.example (hunk 1 above)
Comment (abarghoud) [reply apps/api/.env.example:5]: It's just that I think, when someone want to work on the project he would not use something else as environment than the local one 😉

**Pull request merged**